table.go

```diff
@@ -1092,22 +1094,59 @@ func (t *Table) close() {
 	t.active.pendingWritersWg.Wait()
 	t.closing = true
 	t.active.index.WaitForPendingCompactions()
+
+	for _, closer := range t.closers {
+		if err := closer.Close(); err != nil {
+			level.Error(t.logger).Log("msg", "table closer", "err", err)
+		}
+	}
+}
+
+type CompactionType int
+
+const (
+	CompactionTypeUnknown CompactionType = iota
+	CompactionTypeParquet
+	CompactionTypeParquetDisk
+)
+
+type IndexConfig struct {
+	Level   int
+	MaxSize int64
+	Type    CompactionType
 }
 
 // configureLSMLevels configures the level configs for this table.
-func (t *Table) configureLSMLevels(levels []*index.LevelConfig) []*index.LevelConfig {
+func (t *Table) configureLSMLevels(levels []*IndexConfig) []*index.LevelConfig {
 	config := make([]*index.LevelConfig, 0, len(levels))
 
+	// Check if file compaction is configured, and if so, create a file compaction struct.
+	var fileCompaction *fileCompaction
 	for _, level := range levels {
-		config = append(config, &index.LevelConfig{
-			Level:   level.Level,
-			MaxSize: level.MaxSize,
-			Compact: t.parquetCompaction, // NOTE: right now we only support parquet compaction but in the future this could compact parts into larger arrow records.
-		})
+		if level.Type == CompactionTypeParquetDisk {
+			fileCompaction = t.parquetFileCompaction(len(levels) - 1) // Create a file compaction for all but the last level.
+			t.closers = append(t.closers, fileCompaction)             // Append to closers so that thr underlying files are closed on table close.
+			break
```

Comment:
nit: is this break necessary?

Reply:
Yes, otherwise we may accidentally add multiple closers when we only want one.

Reply:
However clearly I made this more complicated than it needed to be. I change the `fileCompaction` struct to only handle a single level, and just had the configure lsm function create multiple of them.